docs/asciidoc/static/deploying.asciidoc

```diff
@@ -0,0 +1,145 @@
+resource constraints.
+
+image::images/deploy_3.png[]
+
+// [float]
+[[deploying-larger-cluster]]
+=== Scaling to a Larger Elasticsearch Cluster
+
+Typically, Logstash does not communicate with a single Elasticsearch node, but with a cluster made up of several nodes. 
+Logstash can use any of the protocols Elasticsearch makes available to move data into the cluster: 
+{guide}_transport_client_versus_node_client.html[HTTP, transport, or node]. 
+
+You can use the HTTP REST APIs that Elasticsearch provides to index data into the Elasticsearch cluster, transporting 
+the data in a JSON representation. Using the REST APIs does not require the Java client classes or any additional JAR 
+files and has no performance disadvantages compared to the transport or node protocols. You can secure communications 
+that use the HTTP REST APIs with the {shield}[Shield] plugin, which supports SSL and HTTP basic authentication.
+
+You can also use the Java APIs that Elasticsearch provides to serialize the data into a binary representation, using 
+the transport protocol. The transport protocol has the option to sniff the endpoint of the request, selecting an 
+arbitrary client or data node in the Elasticsearch cluster. 
+
+Both the HTTP and transport protocols separate Logstash from the Elasticsearch cluster. The node protocol, by contrast, 
+has the machine running the Logstash instance join the Elasticsearch cluster, running an Elasticsearch instance. The 
+data that needs indexing propagates from this node to the rest of the cluster. Since the machine is part of the 
+cluster, the cluster topology is available, making the node protocol a good fit for use cases that use a relatively 
+small number of persistent connections.
+
+When you use the HTTP protocol, you can configure the Logstash Elasticsearch output plugin to automatically 
+load-balance indexing requests across a 
+{logstash}plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-host[specified set of hosts] in the 
+Elasticsearch cluster. Specifying multiple Elasticsearch nodes also provides high availability for the Elasticsearch 
+cluster by routing traffic to active Elasticsearch nodes.
+You can also use a third-party load balancer, either hardware or software, to handle connections between Logstash and 
+external applications.
+
+NOTE: Make sure that your Logstash configuration does not connect directly to Elasticsearch 
+{ref}modules-node.html[master nodes]. Master nodes tend to have substantial demands on their computing resources. 
+Connect to client or data nodes to maintain cluster performance.
+
+image::images/deploy_4.png[]
+
+// [float]
+[[deploying-message-queueing]]
+=== Managing Throughput Spikes with Message Queueing
+
+When a Logstash pipeline produces data faster than the Elasticsearch cluster can ingest, use a message queue to serve 
+as a buffer. By default, Logstash propagates backpressure to throttle incoming events when indexer consumption rates 
```

Comment:
propagates backpressure? Eww. There's a discussion about how this works elsewhere, here I'd just say "Logstash throttles incoming events..."